ktor-server/ktor-server-core/nix/src/ServerEngineUtils.nix.kt

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.engine
+
+import kotlinx.cinterop.*
+import platform.posix.*
+
+@OptIn(UnsafeNumber::class, ExperimentalForeignApi::class)
+internal actual val WORKING_DIRECTORY_PATH: String get() = memScoped {
+    val result = allocArray<ByteVar>(512)
+    getcwd(result, 512.convert())
```

Comment:
it's splitted per nix and mingwX64 because `getcwd` second argument is Int in place and UInt in other and commonizer don't support it

Reply:
we could make expect-actual for this, but it can be done as housekeeping

Reply:
In PR with commonization of server I replaced it with `kotlinx.io.files` API in `common` code - [`SystemFileSystem.resolve(Path(".")).toString()`](https://github.com/ktorio/ktor/blob/5bd92a11beb8e775fc0de32f27d0e213c714f8ec/ktor-server/ktor-server-core/common/src/io/ktor/server/engine/ServerEngineUtils.kt)
Feel free to comment there on this, as I'm not fully sure, that it's safe to use `kotlinx.io.files` API in `ktor`